### PR TITLE
Added xDevSM tutorial

### PR DESCRIPTION
**Pull request by aferaudo**
### xDevSM tutorial

@Thecave3 Missing paper reference: it will be added once published.

### CODE REVIEW

File: _posts/tutorials/2024-09-16-xdevsm-tutorial.md [hunk 1]
```diff
@@ -0,0 +1,205 @@
+---
+layout: post
+title: "Deploying an xDevSM-Based xApp with OSC NearRT-RIC"
```
Comment (mychele) [inline _posts/tutorials/2024-09-16-xdevsm-tutorial.md:3]:
```suggestion
title: "Deploying an xDevSM-Based xApp with the OSC NearRT-RIC"

```

File: _posts/tutorials/2024-09-16-xdevsm-tutorial.md [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+
+## Prerequisites
+
+Before you begin, ensure you have the following prerequisites installed and configured:
+
+- Docker;
+- Helm;
+- OSC Near-RT RIC:  Open-RAN Software Community (OSC) Near Real-Time Radio Intelligent Controller (RIC) **release-j**.
+
+These tools are necessary for building and creating Helm charts for your xApp.
```
Comment (mychele) [inline _posts/tutorials/2024-09-16-xdevsm-tutorial.md:23]: Any specific version of helm and docker? I would also link to the guide to deploy the OSC RIC, just to give credits to the part below

File: _posts/tutorials/2024-09-16-xdevsm-tutorial.md [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+make
+```
+
+Locate the service model shared library and copy it into your xDevSM project:
+```bash
+find . -type f -name "*.so"
+cp ./src/sm/kpm_sm/kpm_sm_v02.03/libkpm_sm.so xDevSM/sm_framework/lib 
+```
+
+> Note: Details about why we are using ```flexric``` can be found in the paper.
```
Comment (mychele) [inline _posts/tutorials/2024-09-16-xdevsm-tutorial.md:125]: Remember to add link later

**Review by mychele — CHANGES_REQUESTED** [review]
I would add a couple of examples at the end of how you use it with OAI and srs, thoughts?

File: _posts/tutorials/2024-09-16-xdevsm-tutorial.md (hunk 2 above)
Comment (aferaudo) [reply _posts/tutorials/2024-09-16-xdevsm-tutorial.md:23]: I'll add just the helm version.

I'll also include the link to the official OSC RIC deployment procedure.

**Comment by aferaudo** [comment]
It's just a matter of configuration on the SRS side.
About OAI,  I can provide the guide for building the gNB with the E2 Agent and instructions on how to modify the configuration file to enable the connection with the near-RT RIC.

Should we include anything else?

**Comment by Thecave3** [comment]
Does it make sense to have the page in the O-RAN framework as well?

**Comment by aferaudo** [comment]
That makes sense to me. I would focus on the tutorial for this PR and add the framework later if you agree.

File: _posts/tutorials/2024-09-16-xdevsm-tutorial.md [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+title: "Deploying an xDevSM-Based xApp with OSC NearRT-RIC"
+date: 2024-09-16
+category: tutorials
+author: 
+short-description: Guide for deploying an xApp with xDevSM and examples of the effective usage of the xDevSM framework.
+---
+
+This guide provides an overview of deploying an xApp based on the xDevSM framework and offers examples of how to use it effectively. For detailed information about the framework's structure, refer to the paper:
+
+> Add paper
```
Comment (leonardobonati) [inline _posts/tutorials/2024-09-16-xdevsm-tutorial.md:12]: Does it make sense to put the paper at the end of the page, as done in ns-O-RAN, for example? (Up for discussion) https://openrangym.com/tutorials/ns-o-ran

**Comment by leonardobonati** [comment]
@aferaudo if you create an entry for the paper at the end of the `publications` item in [this](https://github.com/aferaudo/openrangym/blob/main/_data/publications.yml) page, it will show in the publications page as well. Then, you can reuse the entry in the page.

**Comment by aferaudo** [comment]
> @aferaudo if you create an entry for the paper at the end of the `publications` item in [this](https://github.com/aferaudo/openrangym/blob/main/_data/publications.yml) page, it will show in the publications page as well. Then, you can reuse the entry in the tutorial page.

@leonardobonati Yes, I saw that page. I didn't include it because we don’t have a link available yet. I can add an entry with random data for the `pdflink` and `bibtexlink` , we can update it once the published version is ready. Alternatively, we can upload it on arxiv and use that for now.

File: _posts/tutorials/2024-09-16-xdevsm-tutorial.md (hunk 4 above)
Comment (aferaudo) [reply _posts/tutorials/2024-09-16-xdevsm-tutorial.md:12]: Yes, we can put it at the end of the page.

**Comment by leonardobonati** [comment]
> > @aferaudo if you create an entry for the paper at the end of the `publications` item in [this](https://github.com/aferaudo/openrangym/blob/main/_data/publications.yml) page, it will show in the publications page as well. Then, you can reuse the entry in the tutorial page.
> 
> @leonardobonati Yes, I saw that page. I didn't include it because we don’t have a link available yet. I can add an entry with random data for the `pdflink` and `bibtexlink` , we can update it once the published version is ready. Alternatively, we can upload it on arxiv and use that for now.

You can leave it empty if you don't have it yet. Please don't add random stuff :-)

**Review by mychele — APPROVED** [review]
Thanks @aferaudo

File: _data/publications.yml [hunk 5]
```diff
@@ -257,6 +257,16 @@ publications:
...
     bibtexlink: https://ece.northeastern.edu/wineslab/wines_bibtex/bonati2024twinning.txt
+  
+  - name: feraudo2024xDevSM
+    author: A. Feraudo, S. Maxenti, A. Lacava, P. Bellavista,  M. Polese, and T. Melodia
+    title: "xDevSM: Streamlining xApp Development With a Flexible Framework for O-RAN E2 Service Models"
+    booktitle: Proceedings of the 18th ACM Workshop on Wireless Network Testbeds, Experimental evaluation & Characterization (WiNTECH)
+    address: Washington, D.C., USA
+    month: November
+    year: 2024
+    pdflink: ""
```
Comment (leonardobonati) [inline _data/publications.yml:268]:
```suggestion
```

File: _data/publications.yml [hunk 6]
```diff
@@ -257,6 +257,16 @@ publications:
...
+    bibtexlink: ""
```
Comment (leonardobonati) [inline _data/publications.yml:269]:
```suggestion
```

File: _data/publications.yml (hunk 5 above)
Comment (leonardobonati) [reply _data/publications.yml:268]: I would just remove this if you don't have an entry yet.

File: _data/publications.yml (hunk 6 above)
Comment (leonardobonati) [reply _data/publications.yml:269]: Why don't you put this? https://ece.northeastern.edu/wineslab/wines_bibtex/feraudo2024xDevSM.txt

**Review by leonardobonati — APPROVED** [review]

**Review by mychele — APPROVED** [review]

**Pull request merged**